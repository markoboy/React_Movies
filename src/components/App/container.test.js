import { render } from '@testing-library/react';
import React from 'react';
import AppContainer from './container';

test('should render Hello React!', () => {
  const component = render(<AppContainer />);

  const element = component.getByText(/hello react!/i);

  expect(element.textContent).toBe('Hello React!');
});
