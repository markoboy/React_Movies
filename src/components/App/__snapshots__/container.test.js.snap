// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container renders a spinner when status is LOADING_STATUS 1`] = `
<div>
  <header
    class="header-nav-container"
  >
    <div
      class="header-nav"
    >
      <div
        class="header-nav__logo-container"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
      <div
        class="header-nav__btn-container"
      >
        <button
          class="btn btn--secondary btn--with-icon"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus fa-w-14 fa-xs "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
          Add Movie
        </button>
      </div>
    </div>
  </header>
  <div
    class="spinner"
  >
    <div
      class="spinner__circle"
    />
    <p>
      Loading...
    </p>
  </div>
  <footer
    class="text-align--center"
  >
    <section
      class="section background--grey"
    >
      <div
        class="wrapper"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
    </section>
  </footer>
  <div
    class="spinner"
  >
    <div
      class="spinner__circle"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`App Container renders components when status is COMPLETE_STATUS 1`] = `
<div>
  <header
    class="header-nav-container"
  >
    <div
      class="header-nav"
    >
      <div
        class="header-nav__logo-container"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
      <div
        class="header-nav__btn-container"
      >
        <button
          class="btn btn--secondary btn--with-icon"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus fa-w-14 fa-xs "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
          Add Movie
        </button>
      </div>
    </div>
  </header>
  <main
    class="site-container__main flex flex--column"
    style="display: none;"
  >
    <div
      class="background--black flex flex--column flex--grow"
    >
      <div
        class="wrapper"
      >
        <ul
          class="result-filter flex"
        >
          <li
            class="result-filter__item result-filter__item--active"
          >
            All
          </li>
          <li
            class="result-filter__item"
          >
            Documentary
          </li>
          <li
            class="result-filter__item"
          >
            Comedy
          </li>
          <li
            class="result-filter__item"
          >
            Horror
          </li>
          <li
            class="result-filter__item"
          >
            Crime
          </li>
          <li
            class="result-filter__item result-filter__item--last"
          >
            <div
              class="result-filter__select-wrapper"
            >
              <label
                class="result-filter__sort-label"
                for="sort-options"
              >
                Sort by
              </label>
              <span
                class="result-filter__select-value"
              >
                Release Date
              </span>
              <select
                class="result-filter__select"
                id="sort-options"
                name="sort"
              >
                <option
                  value="release_date"
                >
                  Release Date
                </option>
                <option
                  value="vote_average"
                >
                  Rating
                </option>
              </select>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </main>
  <div
    class="spinner"
  >
    <div
      class="spinner__circle"
    />
    <p>
      Loading...
    </p>
  </div>
  <footer
    class="text-align--center"
  >
    <section
      class="section background--grey"
    >
      <div
        class="wrapper"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
    </section>
  </footer>
</div>
`;

exports[`App Container renders modal when status is COMPLETE_STATUS and modalIsOpened is true 1`] = `
<div>
  <header
    class="header-nav-container"
  >
    <div
      class="header-nav"
    >
      <div
        class="header-nav__logo-container"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
      <div
        class="header-nav__btn-container"
      >
        <button
          class="btn btn--secondary btn--with-icon"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus fa-w-14 fa-xs "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
          Add Movie
        </button>
      </div>
    </div>
  </header>
  <main
    class="site-container__main flex flex--column"
    style="display: none;"
  >
    <section
      class="banner"
    >
      <picture>
        <img
          alt="Posts of movies on the wall"
          src=""
        />
      </picture>
      <div
        class="banner__content"
      >
        <div
          class="wrapper"
        >
          <h1
            class="banner__title"
          >
            Find your Movie
          </h1>
          <form>
            <div
              class="row"
            >
              <div
                class="column column--m-9"
              >
                <div
                  class="form-element-wrapper"
                >
                  
                  <input
                    class="form__input"
                    id=""
                    name="query"
                    placeholder="What do you want to watch?"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="column column--m-3"
              >
                <div
                  class="form-element-wrapper"
                >
                  <button
                    class="btn btn--primary btn--full-width"
                    tabindex="0"
                    type="submit"
                  >
                    Search
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </section>
    <div
      class="background--black flex flex--column flex--grow"
    >
      <div
        class="wrapper"
      >
        <ul
          class="result-filter flex"
        >
          <li
            class="result-filter__item result-filter__item--active"
          >
            All
          </li>
          <li
            class="result-filter__item"
          >
            Documentary
          </li>
          <li
            class="result-filter__item"
          >
            Comedy
          </li>
          <li
            class="result-filter__item"
          >
            Horror
          </li>
          <li
            class="result-filter__item"
          >
            Crime
          </li>
          <li
            class="result-filter__item result-filter__item--last"
          >
            <div
              class="result-filter__select-wrapper"
            >
              <label
                class="result-filter__sort-label"
                for="sort-options"
              >
                Sort by
              </label>
              <span
                class="result-filter__select-value"
              >
                Release Date
              </span>
              <select
                class="result-filter__select"
                id="sort-options"
                name="sort"
              >
                <option
                  value="release_date"
                >
                  Release Date
                </option>
                <option
                  value="vote_average"
                >
                  Rating
                </option>
              </select>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </main>
  <div
    class="spinner"
  >
    <div
      class="spinner__circle"
    />
    <p>
      Loading...
    </p>
  </div>
  <footer
    class="text-align--center"
  >
    <section
      class="section background--grey"
    >
      <div
        class="wrapper"
      >
        <h1
          class="brand-logo"
        >
          <a
            href="/"
          >
            netflix
            <span
              class="font--regular"
            >
              roulette
            </span>
          </a>
        </h1>
      </div>
    </section>
  </footer>
</div>
`;
